react/src/hooks/useMemoWithPrevious.tsx

```diff
@@ -0,0 +1,36 @@
+import { DependencyList, useEffect, useMemo, useRef, useState } from 'react';
+
+export const useMemoWithPrevious = <T,>(
+  factory: () => T,
+  deps: DependencyList,
+  { initialPrev }: { initialPrev?: T } | undefined = {},
+) => {
+  const prevRef = useRef(initialPrev);
+  const [prevResetKey, setPrevResetKey] = useState({});
+
+  const current = useMemo(factory, deps);
```

Comment:
https://github.com/lablup/backend.ai-webui/actions/runs/11698681318/job/32579369720?pr=2820#step:9:10

Reply:
I don't know why it was deleted. 😢 I just added it again.